File: baselib/src/main/java/com/stjy/baselib/wigiet/photopreview/activity/PhotoPreviewActivity.java
package com.stjy.baselib.wigiet.photopreview.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;

import com.blankj.utilcode.util.BarUtils;
import com.stjy.baselib.R;
import com.stjy.baselib.base.mvvm.BaseActivity;
import com.stjy.baselib.wigiet.photopreview.adapter.PhotoPagerAdapter;
import com.stjy.baselib.wigiet.photopreview.fragment.PhotoFragment;

import java.util.ArrayList;
import java.util.List;
/**
 * @author acb
 */
public class PhotoPreviewActivity extends BaseActivity {

    public static final String POSITION = "BASICRES_POSITION";
    public static final String IMAGES = "BASICRES_IMAGE";
    private List<String> images;
    private int mPosition;
    private int size;
    private ViewPager vpPhoto;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_photo_preview);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void initData() {

    }

    @Override
    public void initStatusBar() {
        BarUtils.setStatusBarColor(this, ContextCompat.getColor(this,R.color.black));
        BarUtils.addMarginTopEqualStatusBarHeight(getToolbar());
        BarUtils.setStatusBarLightMode(this, false);
    }

    @Override
    public void initView() {
        getToolbar().setBackgroundResource(R.color.black);
        vpPhoto = findViewById(R.id.vpPhoto);
        Bundle bundle = getIntent().getExtras();
        mPosition = bundle.getInt(POSITION);
        images = bundle.getStringArrayList(IMAGES);
        if (images != null) {
            this.size = images.size();
            List<Fragment> fragments = new ArrayList<>();
            for (int i = 0; i < images.size(); i++) {
                fragments.add(PhotoFragment.newInstance(images.get(i)));
            }
            PhotoPagerAdapter photoPagerAdapter = new PhotoPagerAdapter(getSupportFragmentManager(), fragments);
            vpPhoto.setAdapter(photoPagerAdapter);
            vpPhoto.setCurrentItem(mPosition);
            setBarTitle(mPosition + 1 + "/" + size, ContextCompat.getColor(this, R.color.white));
        }
    }

    @Override
    public void initListener() {
        vpPhoto.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                setBarTitle(position + 1 + "/" + size);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }

    /**
     * @param context
     * @param position 当前图片位置
     * @param images   图片集合
     */
    public static void actionStart(Context context, int position, List<String> images) {
        Intent intent = new Intent(context, PhotoPreviewActivity.class);
        Bundle bundle = new Bundle();
        bundle.putInt(POSITION, position);
        bundle.putStringArrayList(IMAGES, (ArrayList<String>) images);
        intent.putExtras(bundle);
        context.startActivity(intent);
    }
}

File: baselib/src/main/java/com/stjy/baselib/wigiet/photopreview/activity/PhotoPreviewActivity.kt
package com.stjy.baselib.wigiet.photopreview.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewPager
import android.support.v4.view.ViewPager.OnPageChangeListener
import com.blankj.utilcode.util.BarUtils
import com.stjy.baselib.R
import com.stjy.baselib.base.mvc.BaseActivity
import com.stjy.baselib.wigiet.photopreview.adapter.PhotoPagerAdapter
import com.stjy.baselib.wigiet.photopreview.fragment.PhotoFragment.Companion.newInstance
import java.util.*

/**
 * @author acb
 */
class PhotoPreviewActivity : BaseActivity() {
    private lateinit var images: List<String>
    private lateinit var vpPhoto: ViewPager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_photo_preview)
    }

    public override fun initData() {}
    override fun initStatusBar() {
        BarUtils.setStatusBarColor(this, ContextCompat.getColor(this, R.color.black))
        BarUtils.addMarginTopEqualStatusBarHeight(toolbar!!)
        BarUtils.setStatusBarLightMode(this, false)
    }

    public override fun initView() {
        val bundle = intent.extras
        images = bundle.getStringArrayList(IMAGES)?: arrayListOf()
        toolbar?.setBackgroundResource(R.color.black)
        vpPhoto = findViewById(R.id.vpPhoto)

        val fragments: MutableList<Fragment> = ArrayList()
        for (i in images.indices) {
            fragments.add(newInstance(images[i]))
        }
        val photoPagerAdapter = PhotoPagerAdapter(supportFragmentManager, fragments)
        vpPhoto.adapter = photoPagerAdapter
        vpPhoto.currentItem = bundle.getInt(POSITION)
        setBarTitle("${vpPhoto.currentItem + 1}/${images.size}", ContextCompat.getColor(this, R.color.white))
    }

    public override fun initListener() {
        vpPhoto.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                setBarTitle("${vpPhoto.currentItem + 1}/${images.size}")
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
    }

    companion object {
        const val POSITION = "BASICRES_POSITION"
        const val IMAGES = "BASICRES_IMAGE"

        /**
         * @param context
         * @param position 当前图片位置
         * @param images   图片集合
         */
        fun start(context: Context, position: Int, images: List<String?>?) {
            context.startActivity(with(Intent(context, PhotoPreviewActivity::class.java)){
                this.putExtras(with(Bundle()){
                    this.putInt(POSITION, position)
                    this.putStringArrayList(IMAGES, images as ArrayList<String?>?)
                    this
                })
                this
            })
        }

        /**
         * @param context
         * @param position 当前图片位置
         * @param image   图片路径或url
         */
        fun start(context: Context, position: Int, image: String?) {
            context.startActivity(with(Intent(context, PhotoPreviewActivity::class.java)){
                this.putExtras(with(Bundle()){
                    this.putInt(POSITION, position)
                    this.putStringArrayList(IMAGES, object : ArrayList<String?>() {
                        init {
                            add(image)
                        }
                    })
                    this
                })
                this
            })
        }
    }
}